processor/processor.go

```diff
@@ -1820,9 +1820,16 @@ func (proc *Handle) processJobsForDest(partition string, subJobs subJob) *transf
 				&backendconfig.DestinationT{},
 			)
 
-			// TODO: TP ID preference 1.event.context set by rudderTyper   2.From WorkSpaceConfig (currently being used)
-			shallowEventCopy.Metadata.TrackingPlanId = source.DgSourceTrackingPlanConfig.TrackingPlan.Id
-			shallowEventCopy.Metadata.TrackingPlanVersion = source.DgSourceTrackingPlanConfig.TrackingPlan.Version
+			trackingPlanId := source.DgSourceTrackingPlanConfig.TrackingPlan.Id
+			trackingPlanVersion := source.DgSourceTrackingPlanConfig.TrackingPlan.Version
+			rudderTyperTPID := misc.MapLookup(singularEvent, "context", "ruddertyper", "trackingPlanId")
+			rudderTyperTPVersion := misc.MapLookup(singularEvent, "context", "ruddertyper", "trackingPlanVersion")
+			if rudderTyperTPID != nil && rudderTyperTPVersion != nil {
```

Comment:
Why do we need to check for both the values and then set both the values? Can we check the validity for each fields independently and overwrite it?
```GO
			rudderTyperTPID := misc.MapLookup(singularEvent, "context", "ruddertyper", "trackingPlanId")
			if rudderTyperTPID != nil {
				if id, ok := rudderTyperTPID.(string); ok {
					trackingPlanId = id
				}
			}
			rudderTyperTPVersion := misc.MapLookup(singularEvent, "context", "ruddertyper", "trackingPlanVersion")
			if rudderTyperTPVersion != nil {
				if version, ok := rudderTyperTPVersion.(int); ok {
					trackingPlanVersion = version
				}
			}
```

Reply:
Using tracking planId from workspaceConfig and version from rudder typer or vice versa is error prone. Only if both values are present, we will go rudder typer values